upgrade/change_v21.03.0.go

```diff
@@ -0,0 +1,449 @@
+		}
+		pl := pb.PostingList{}
+		if err := pl.Unmarshal(val); err != nil {
+			return err
+		}
+		for _, p := range pl.Postings {
+			corsVals = append(corsVals, p.Value)
+		}
+		return nil
+	})
+	return corsVals, err
+}
+
+func dropDepreciated(db *badger.DB) error {
+	var prefixes [][]byte
+	for pred := range deprecatedPreds {
+		pred = x.GalaxyAttr(pred)
+		prefixes = append(prefixes, x.SchemaKey(pred), x.PredicatePrefix(pred))
+	}
+	for typ := range deprecatedTypes {
+		prefixes = append(prefixes, x.TypeKey(x.GalaxyAttr(typ)))
+	}
+	return db.DropPrefix(prefixes...)
+}
+
+// fixCors removes the internal predicates from the restored backup. This also appends CORS from
+// dgraph.cors to GraphQL schema.
+func fixCors(db *badger.DB) error {
+	glog.Infof("Fixing cors information in the restored backup.")
+	cors, err := getCors(db)
+	if err != nil {
+		return errors.Wrapf(err, "Error while getting cors from db.")
+	}
+	if err := updateGqlSchema(db, cors); err != nil {
+		return errors.Wrapf(err, "Error while updating GraphQL schema.")
+	}
+	return errors.Wrapf(dropDepreciated(db), "Error while dropping depreciated preds/types.")
+}
+
+// fixPersistedQuery, for the schema related to persisted query removes the deprecated field from
+// the type and updates the index tokenizer for the predicate.
+func fixPersistedQuery(db *badger.DB) error {
+	glog.Infof("Fixing persisted query schema in restored backup.")
+	update := func(entry *badger.Entry) error {
+		txn := db.NewTransactionAt(math.MaxUint64, true)
+		defer txn.Discard()
+		if err := txn.SetEntry(entry); err != nil {
+			return err
+		}
+		// Schema is written at version 1.
+		return txn.CommitAt(1, nil)
+	}
+
+	// Update the tokenizer in the schema.
+	su := pb.SchemaUpdate{
+		Predicate: x.GalaxyAttr("dgraph.graphql.p_query"),
+		ValueType: pb.Posting_STRING,
+		Directive: pb.SchemaUpdate_INDEX,
+		Tokenizer: []string{"sha256"},
+	}
+	data, err := su.Marshal()
+	if err != nil {
+		return err
+	}
+	entry := &badger.Entry{}
+	entry.Key = x.SchemaKey(x.GalaxyAttr("dgraph.graphql.p_query"))
+	entry.Value = data
+	entry.UserMeta = posting.BitSchemaPosting
+	if err := update(entry); err != nil {
+		return errors.Wrap(err, "while updating persisted query's schema")
+	}
+
+	// Update the type.
+	tu := pb.TypeUpdate{
+		TypeName: x.GalaxyAttr("dgraph.graphql.persisted_query"),
+		Fields:   []*pb.SchemaUpdate{&su},
+	}
+	data, err = tu.Marshal()
+	if err != nil {
+		return err
+	}
+	entry = &badger.Entry{}
+	entry.Key = x.TypeKey(x.GalaxyAttr("dgraph.graphql.persisted_query"))
+	entry.Value = data
+	entry.UserMeta = posting.BitSchemaPosting
+	if err := update(entry); err != nil {
+		return errors.Wrap(err, "while updating persisted query's type")
+	}
+	return nil
+}
+
+// UpgradeFrom2011To2103 upgrades a p directory restored from backup of 20.11 to the changes in
+// 21.03. It fixes the cors, schema and drops the deprecated types/predicates.
+func UpgradeFrom2011To2103(db *badger.DB) error {
```

Comment:
just a thought. i guess calling it online/offline makes more sense

Reply:
Yeah right. Done.